src/views/Unfollow/index.tsx

```diff
@@ -0,0 +1,118 @@
+import React, { useEffect } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory, useLocation } from 'react-router-dom';
+import { useMutation, useQuery } from '@apollo/client';
+
+import Alert from 'components/shared/Alert';
+import useMessage from 'hooks/useMessage';
+import GetBasics from 'queries/Basics/GetBasics';
+import {
+  GetBasics as GetBasicsType,
+  GetBasicsVariables
+} from 'queries/Basics/types/GetBasics';
+import DeletePlanFavorite from 'queries/Favorite/DeletePlanFavorite';
+import { DeletePlanFavoriteVariables } from 'queries/Favorite/types/DeletePlanFavorite';
+
+type UnfollowProps = {
+  children: React.ReactNode;
+};
+
+const UnfollowWrapper = ({ children }: UnfollowProps) => {
+  const history = useHistory();
+  const { t } = useTranslation('modelPlan');
+  const { pathname, search } = useLocation();
+  const { showMessageOnNextPage } = useMessage();
+
+  const modelIDToRemove = search
+    .replace(/.*modelID=(.*)\/?/g, '$1')
+    .replace(/\/+$/, '');
+
```

Comment:
This param extraction is natively built in with `URLSearchParams` and can be used like so:

```
const params = new URLSearchParams(search);
const modelIDToRemove = params.get('modelID');
```